### PR TITLE
feat(SEASON): controller, entity, service

### PR DESCRIPTION
**Pull request by Rossb0b**

### CODE REVIEW

File: src/modules/season/dto/update-season.dto.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { PartialType } from '@nestjs/swagger';
+import { CreateSeasonDto } from './create-season.dto';
+
+export class UpdateSeasonDto extends PartialType(CreateSeasonDto) {
+  name: string;
```
Comment (draphikas) [inline src/modules/season/dto/update-season.dto.ts:5]: Same as previous PR, name is already in CreateDto

File: src/modules/season/dto/update-season.dto.ts [hunk 2]
```diff
@@ -1,6 +1,3 @@
-import { PartialType } from '@nestjs/swagger';
 import { CreateSeasonDto } from './create-season.dto';
 
-export class UpdateSeasonDto extends PartialType(CreateSeasonDto) {
-  name: string;
-}
+export class UpdateSeasonDto extends CreateSeasonDto {}
```
Comment (draphikas) [inline src/modules/season/dto/update-season.dto.ts:3]: Partial is necessary

File: src/modules/season/season.service.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Season } from './entities/season.entity';
+import { CreateSeasonDto } from './dto/create-season.dto';
+import { UpdateSeasonDto } from './dto/update-season.dto';
+
+const kNotFoundError = 'Unable to find the given thematic';
```
Comment (cchampou) [inline src/modules/season/season.service.ts:8]: `NotFoundError`  instead no ?

File: src/modules/season/season.service.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    return season;
+  }
+
+  async update(id: number, updatedSeason: UpdateSeasonDto) {
+    const season = await this.findOne(id);
+
+    if (!season) {
+      throw new BadRequestException(kNotFoundError);
```
Comment (cchampou) [inline src/modules/season/season.service.ts:42]: `NotFoundException` instead :)

File: src/modules/season/season.service.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    await this.seasonRepository.save(season);
+
+    return season;
+  }
+
+  async remove(id: number) {
+    const season = await this.findOne(id);
+
+    if (!season) {
+      throw new BadRequestException(kNotFoundError);
```
Comment (cchampou) [inline src/modules/season/season.service.ts:55]: `NotFoundException` instead :wink:

**Review by cchampou — APPROVED** [review]
👍

**Pull request merged**